proto/backend.proto

```diff
@@ -306,3 +306,164 @@ message StreamPacket {
   // JSON-encoded data to publish into a channel.
   bytes data = 1;
 }
+
+//-----------------------------------------------------------------
+// InstanceSettings -- EXPERIMENTAL and is subject to change until 12.??
+//-----------------------------------------------------------------
+
+service InstanceSettings {
+  // This allows verifying the app/datasource settings before saving
+  // This is a specalized form of the validation/mutation hooks that only work for instance settings
+  // The response will be in the preferred storage apiVersion
+  rpc ProcessInstanceSettings(ProcessInstanceSettingsRequest) returns (ProcessInstanceSettingsResponse);
+}
+
+message ProcessInstanceSettingsRequest {
```

Comment:
Where are the settings that we're processing?

Reply:
In the case of a mutation, will the previous and new settings both be included?

Reply:
https://github.com/grafana/grafana/pull/87718/files#diff-e75b2165866a30555409a61945b14cb5ab7b8270b5b4db8b45759a0e606014e4R295

For this use case, do we need the previous settings?  

In addition to this endpoint (and when I originally requested your review!) I was also trying to have a generic resource callback.  After looking more, I want to try that with the upstream k8s proto defs rather than defining our own.  However for validating the existing instance settings I think using the existing proto as much as possible makes sense.